### ISSUE TITLE
pbtranscript-tofu

### ISSUE THREAD

**Issue by jxu006** [PacificBiosciences/cDNA_primer#1]
Hi Liz, 

I downloaded the pbtranscript-tofu and there are 3 folders in the directory: /cDNA_primer/pbtranscript-tofu

pbtranscript
pbtranscript_20150106_forYli
pbtranscript_old

Which one should I use? 

Thanks,

**Comment by Magdoll** [comment]
Please use pbtranscript/. The other two are obsolete and have been removed.

**Comment by jxu006** [comment]
Hi Liz,

Do you have a tutorial on how to install pbtranscript-tofu and how to run
it? I have a little difficulty in installing it.

Thanks,

Jack

**Comment by Magdoll** [comment]
Yes, the installation is here:
https://github.com/PacificBiosciences/cDNA_primer/wiki/Setting-up-virtualenv-and-installing-pbtranscript-tofu

additional tutorials are listed in the wiki:
https://github.com/PacificBiosciences/cDNA_primer/wiki#tofu

**Comment by jxu006** [comment]
Thanks a lot, Liz.

I have another question:
If I run the pbtranscript classify and pbtranscript cluster, I do not need
the pbtranscript-tofu, correct?

If I want to run collapse_isoforms_by_sam.py, I will need the tofu,
correct?

Thanks,

**Comment by Magdoll** [comment]
Hi Jack,

Yes if you already ran pbtranscript.py "classify" and "cluster", you do not need TOFU.

And yes, collpase_isoforms_by_sam.py is only available in TOFU now.

Some users had trouble running collapse script with an import failure. Today I identified the problem but was only able to fix it in a hacky manner. Please see [issue 3](https://github.com/PacificBiosciences/cDNA_primer/issues/3). If you encounter the same error, try to fix it using the fix mentioned.

**Comment by jxu006** [comment]
Hi Liz,

I ran the ConsensusTools.sh using the command line:

ConsensusTools.sh CircularConsensus  --minFullPasses 0
--minPredictedAccuracy 75 --parameters
/global/projectb/shared/software/smrtanalysis/2.3.0/current/analysis/etc/algorithm_paramete

rs/2014-09 --numThreads 24 --fofn input.fofn -o output



Then I go to output folder:

jianpeng@gpint100:/global/scratch2/sd/jianpeng/PacBio/PB0952/test_consensusTools/output$
ls -l

total 120064

-rw-r----- 1 jianpeng jianpeng  715621 Mar 16 12:01
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.1.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 1362623 Mar 16 12:01
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.1.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 3442815 Mar 16 12:01
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.1.ccs.h5

-rw-r----- 1 jianpeng jianpeng  813482 Mar 16 12:03
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.2.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 1551575 Mar 16 12:03
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.2.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 3604344 Mar 16 12:03
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.2.ccs.h5

-rw-r----- 1 jianpeng jianpeng  651781 Mar 16 12:05
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.3.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 1242380 Mar 16 12:05
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.3.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 3301232 Mar 16 12:05
m140715_021247_42173_c100632222550000001823124209121482_s1_p0.3.ccs.h5

-rw-r----- 1 jianpeng jianpeng 3757850 Mar 16 12:17
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.1.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 7164192 Mar 16 12:17
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.1.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 9310534 Mar 16 12:17
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.1.ccs.h5

-rw-r----- 1 jianpeng jianpeng 3641055 Mar 16 12:32
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.2.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 6942248 Mar 16 12:32
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.2.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 8773618 Mar 16 12:32
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.2.ccs.h5

-rw-r----- 1 jianpeng jianpeng 3116628 Mar 16 12:44
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.3.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 5934456 Mar 16 12:44
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.3.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 7852104 Mar 16 12:44
m140716_204214_42173_c100631902550000001823124209121400_s1_p0.3.ccs.h5

-rw-r----- 1 jianpeng jianpeng 2332588 Mar 16 12:54
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.1.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 4442326 Mar 16 12:54
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.1.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 6542674 Mar 16 12:54
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.1.ccs.h5

-rw-r----- 1 jianpeng jianpeng 2700792 Mar 16 13:06
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.2.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 5141099 Mar 16 13:06
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.2.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 7234914 Mar 16 13:06
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.2.ccs.h5

-rw-r----- 1 jianpeng jianpeng 3421913 Mar 16 13:19
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.3.ccs.fasta

-rw-r----- 1 jianpeng jianpeng 6512020 Mar 16 13:19
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.3.ccs.fastq

-rw-r----- 1 jianpeng jianpeng 8457565 Mar 16 13:19
m140716_235824_42173_c100631902550000001823124209121401_s1_p0.3.ccs.h5



I did not find the reads_of_insert.fasta output file. Do you know why? It
seems that it finished running.

Did it stop in the middle?


Thanks,

**Comment by Magdoll** [comment]
Hi Jack,

You can create reads_of_insert.fasta just by concatenating the files.

Do:

cat *.ccs.fasta > reads_of_insert.fasta
cat *.ccs.fastq > reads_of_insert.fastq
cat  <full_directory_path>/data/*.ccs.h5 > reads_of_insert.fofn

**Comment by jxu006** [comment]
I see.  Did  the ConsensusTools.sh suppose to generate the file
* reads_of_insert.fasta, **reads_of_insert.fastq and *reads_of_insert.fofn?

Thanks a lot, Liz.

**Comment by Magdoll** [comment]
Hi Jack,

I don't think so. I think the concatenation was originally done as a separate step in the pipeline. Since you are calling ConsensusTools.sh you just have to concatenate them yourself. It is not a bug I believe.

**Comment by jxu006** [comment]
I understand. Thanks, Liz.

I found the webpage:
https://github.com/PacificBiosciences/cDNA_primer/wiki/Tutorial:-Getting-ReadsOfInsert-from-SMRT-Portal-or-through-the-command-line

At the end of this page, it showed:
After the job is completed, you will find the output in
/MYHOME/test_dir/data/reads_of_insert.fasta

I am not sure if this is generated by ConsensusTools.sh or not.

Thanks,

**Comment by Magdoll** [comment]
Hi Jack,

I think it is my fault for not adding to the tutorial about needing to concatenate. I will do that! Thank you for reminding.

**Comment by jxu006** [comment]
Thanks, Liz.

I have another question:
I want to identify the rRNA contamination in my Iso-Seq data. I collect the
rRNA sequences and want to run BLASR to align the reads of insert (ROI) to
rRNA sequences. Is it possible that BLASR reports the alignment with very
low coverage? For example, the reads is 1000 bp in length, and only 100 bp
is mapped to rRNA and BLASR report it as an alignment. Do you know if BLASR
can specify the minimum coverage to report the alignment?

Do you have suggestions on other aligners that are suitable for rRNA
contamination identification?

Thanks a lot,

Jianpeng

**Comment by jxu006** [comment]
Hi Liz,

I tried to run collapse_isoforms_by_sam.py, but it showed the error
message:

Traceback (most recent call last):

  File
"/global/homes/j/jianpeng/tofu/cDNA_primer/bin/collapse_isoforms_by_sam.py",
line 5, in <module>

    pkg_resources.run_script('pbtools.pbtranscript==0.4.1',
'collapse_isoforms_by_sam.py')

  File
"/global/homes/j/jianpeng/tofu/cDNA_primer/lib/python2.7/site-packages/pkg_resources.py",
line 534, in run_script

    self.require(requires)[0].run_script(script_name, ns)

  File
"/global/homes/j/jianpeng/tofu/cDNA_primer/lib/python2.7/site-packages/pkg_resources.py",
line 1434, in run_script

    execfile(script_filename, namespace, namespace)

  File
"/global/u2/j/jianpeng/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/EGG-INFO/scripts/collapse_isoforms_by_sam.py",
line 155, in <module>

    main(args)

  File
"/global/u2/j/jianpeng/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/EGG-INFO/scripts/collapse_isoforms_by_sam.py",
line 108, in main

    check_ids_unique(args.input, is_fq=args.fq)

  File
"/global/homes/j/jianpeng/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/Utils.py",
line 23, in check_ids_unique

    if r.id in seen:

AttributeError: 'FastqRecord' object has no attribute 'id'


My command is:

collapse_isoforms_by_sam.py --input all_quivered_hq.100_30_0.99.fastq --fq
-s all_quivered_hq.100_30_0.99.sorted.sam -o collapse_final_output


Do you know what is going on? How to fix this?


Thanks,


Jack

**Comment by jxu006** [comment]
Hi Liz,

My GMAP output SAM file is something like this:
-------------------------------------------------

@HD VN:1.0 SO:unsorted

@PG ID:GMAP PN:gmap VN:2014-06-10 CL:gmap -D
/global/homes/j/jianpeng/Reference_genome/Athaliana/referenceGenome -d
arabidopsis -f samse -n 0 -t 24 --min-trimmed-coverage 0.99 --min-i

dentity 0.85 all_quivered_hq.100_30_0.99.fastq

@SQ SN:Chr1 LN:30427671

@SQ SN:Chr2 LN:19698289

@SQ SN:Chr3 LN:23459830

@SQ SN:Chr4 LN:18585056

@SQ SN:Chr5 LN:26975502

@SQ SN:ChrC LN:154478

@SQ SN:ChrM LN:366924

c30/f3p78/3143 16 ChrM 8943 40 22M1D1507M1I1250M1D353M1D10M * 0 0
GTTTAGTACGAGATGGCTTCACACCTCGCGGTGCTTCCACCTCTCGCCTATCGAAGTTCTGTTCTAAAACCTGTTCCGAGAACTT

GTATAGAGAAGGATTTCCCGCTAAGCAGCAGTTCTTCCATACCAACTTAGCTGCCCGGCGCTGCTATTGGCATAACAACCGGTACACCATAGGTTGGCCCAACCCAGTCCTCTCGTACTAGGGTTGGCTCCTCGCAGTTCTCCCTTTAACACCAACGGTAGATAGGAACCGAACTGTCTCACGACGTTC

TAAACCCAACTCACGTACCACTTGAATCGGCGAACAACCGAACCCTTGGGACCTTCTTCAACCCCAGGATGTGATGAGTCGACATCGAGGTGCCAAACGACTCCGTCGATAAGAGCTCTTGGGAGTCATCAGCCTGTTATCCCCGGCGTACCTTTGATCCGTTGAGCGAGAGCCCTTCCACACGGGACT

CCCGGATCACTATGGCCGACTTTCGTCTCTGTTCGACCAGTCGGTCTCACAGTCAGGCAGGCTTATACCATTACGCTCACGAGCAGAATCTTAGCTTGAGCCTACCTTCGCACACCTCCGTTACTCTTTAGGAGGCATCCGCCCCAGATAAACTACCCACCTCGCAGTGTCCCGCCTCCCCCCGAATTC

TCGGTGCGGCGGTTAGGCACCCTTAGACGAAAGAGTGGTCTTTCAGGATTGGTCCTTCTATGTCACGACCTCCCACCTATCCTACACATTCGATCAAGGTTGTCACTGCGAAGCTATAGTGAAGGTGCACGGGGTCTTACCGTCTAGCCGTTGGTACTCCGCATCTTCACGGAGAATTCAATTTCACCG

GGTCCATGTCGGAGACAGCGGGGCAGTCGTTACACCATTCGTGCAGGTCGCTACTTATGCGACAAGGAATTTCGCTACCTTAGGACAGTTAGAGTTACTGCCGCCGTTTACCGGGGCTTCCATTCAAAGCTTATAACACTTCTCCTTTTGACTTTCCAGCACCGGGCAGGTGTCAGACTCTATACATCG

TGTTACCACTTAGCAGAGTCCTGTGTTTTTAATAAACAGTCGCTACCCCCTGGTATGTGCCGCTTTCCTAATCAAAAGATAGGAGAGCACCCCTTCTCCCGAAGTTACGGGGTCATTTTGCCGAGTTCCTTCGACATGGTTCTCTCAAGCGCCCTAGTATACTCTACTTGTTCACCTGTGTCGGTTTGG

GGTACGGTCAGTTCACCGGGAGGATCGCCCTCCCAATTCGAAGTTTTTTCCTGGAAGTTTCAACCTTCTTGACTATGACAAGAGTCGCGACTATAAACAGACTCGCGACTATGGCAGGGCGGTACGCTCTGCTCTCTCGCGACCCCTACTCTAATCAAAAGACTAAAGGCCCCTACTGAAGGTCGCCAA

ACTACGACGAGACTTTCGCCTTTTGAAGCGCCAGTAGCGTAGGGCGACCGGGCCAGGCCGAGTCAGAAAGGCTTTGATGACTCAAGGTTCATATTAGGGAAAGGAGAGTGAGGGGAAGAGGCGCTACGCTGCCCTCGGCCCGATCATCCAATTCGCTCCAAGAGAGAGGCATGGTTTTGTAGTCAAAGC

AACTTCGTCACTTTCGTGTACCCATCGGACGGCAGCCCTTTCGGGGGTTCCTTAGGGACCGATTCACTCTGCGTAGATTGACTGAACGCAGAAAGCCTTCCACTGGCAGGCGATCGTGTTTTTCACAGGATTTATCGTTACTCATGTCAGCATTCTCACTTCTGATATCTCCAGGTCTTGTCACCAAAA

ACCTTCCCCGATTGACAGAACGTTCCGCTACTGACACTTGAAAAAGCAGCTTTCAAGGTCTCGTCGCTTCGGTGAATCACTTGAGCCCTGATACATTTTCGGTGCCATGGAGCTAGACCAGTGAGCTATTACGCTTTCTTCAAAGGATGGCTGCTTCCAAGCCCACCTCCTGGTTGTCATCGCTCGATC

ACTTCCTTTTCCACTAAGTGATTGCTTAGGGACCTTAGCGTACGATCTGGGCTGTTTCCCTCTCGACTTTGGATCTTAGCACCCCAAAAGTCTGTCTGTACAAACGAGAACGGCCTGTATTCGGAGTTTCCCTGGGGTTGGTAAGGCGAAATGGGGCCACCCTAGCCCATTGAGTGCTCTACCTCGGGC

CATCGACATCATACGCTCTACTGAAATAGATTTCGCGGAAAACCAGCTATATCCGATCTTGGTTGGCCTTTCACCCCTAGCCACAAGTCATCCCCGTATTTTGCCACATACGTGGGTTCGGTCCTCCAAGGCCTGTTAGAGCTCTCTTCAACCTGCTCATGGCTAGATCGATCGGTTTCGGGTCAAATA

GGAAGAACTAGAAGATTCCACCTTTGGAAAGCGCCTACACCTAATGGCTTAAGCCGCTCTTCCCATTTCCTCGCTGACCCATCATGCAAAAGGTACGCCGTTAGAGTGAGTGCGCTTGACTACTCCTTCGACTGATTGTTCGCATCGGATCTCAGGTTCTCTATTGCACTCCCGTCATAGGGTTCTTTT

CACCTTTCCCTCACGGTACTTGTACGCTATCGGTCATTGAGGAATACTTAGGCTTAGAGGGTGGTCCCCCTTTCTCGCGTAAAAGCGATCAGAATTCGAACACGCCGCGTTTTACTGGGAAGGATCGAACCATAGGAACGAATCTACAGGGCTATCACCTTCTTTGGCCAGATCTTCCAACCTTTTCAC

AATTACAGTTCACAGCGCCCTTTAGGAATCTTCAGTCAAGGTACGAAGTAACTCGACTGAAAGAAGAGGGGCTTGCTGGTTTTTCCATCATCCAATCCACAATAAATCGAATGAAACCTGGCGAAAAAGAAGTGAACACTTTGGAACGAAGCTTCGTCTTTCTTTTTCTTAAAATCCCAAATCCGCTCT

CGCTCGCCGCTACTAACGGGGTCTCGGTTGATTTCCCTTCCTTTAGCTACTTAGATGTTTCAGTTCGCTAAGTTTTCAAAGTCCAAAGAGCGCAGACTAGCCACGGAGCTTGGATACGGTTTCCCGATCGGAGATCCATGGATCACAGACGGTATCTCCCCATGGCCTTTCGCCTCTGAAAGCGTCCTT

CCTTCTCAATGCCCGGGCATCCATCAatgcattc
@QSQPQPQQQRRQRTSQRQRQRRRRRPQQQQPQQTPRPQSQSRQQQTRQQPPQRQSQPRRRRRRQRTRQTSRRTRRTPQRRRQSQRSPPQQQQTQTRRRSSRSRRRSSQQRPQQRRQQQRRSRRQQSRTRRSRPPRPQTSRSSSRRQQQ

QRRQQQQSRQQQQQRQUQQQSRRQQQQQPSQTRRUSSRRRQQSRSRRSRQQQRRRRSRSRRQPQRSPSRQQQQQQQPQRQTRQSSQQSQQQQRPRQQQRSTRSSSRRRQRQRTSQRRQQPQRSSQQTSRSQQRQQQRRRRRQRQQRTRTTQRTTRPSSSQSQSSRRSTSSTRUQPQRRSRRRSQRRUSS

RSRRRRRRRRSSRQRRSQQSRSRRSQQQSRSSQQQQSRSPRQRQSRSRQQQQSQSRRWRPQQQQQQRRRRQRSRSRUQTQTSRRSQUTRSRQRPQQQQQQQQRRQQQQRSRRSSQPQQQSTQQRSSTRSSRRQRRRQQRQRRQQRRRQQQQRRRRQRRRRQRPQRRRQQRQQRRRQQPRPSRRRSQQQQ

SQPRQQQQQQQQQPQRRQQQQRSSRRRRRRQQQSSRQRQQRQQSRRRPQQTQSRQRSRRSRRTQQQRRQQRPRRTRRSSRRPRQSTRQQQQSSRTSRRRSQQQRPTRUSPSRQQQRRQQSQRTSQQQSRRPQRQQQQQRPSQQQRTQQQPPQRQQQQQQQSRQRRSQQPPUPQRRQPSTRRRRSRRRQR

QRRQSPQQSQQSQRRPSPRRQQSQTQQRPRRSRQPRQQRQQPRQQPQRQQRSRRRRQSRSRSQQRRSRSRQQRQQSQSPQRRRSQRTRRQRSQRQQQRSRQTQRQSRQTRUTRQRRQSSRRQUQRRPQRQQQQRQSQSQQRPRQSRSPRRPQRRSRQRQQQQQQQPQQPQTRQQRQQQRRRQRRQQRQR

QSSRQTSRRQQQQRSTRQRRRQPQQRRRRSSQQPQSRRQRQQTQQRTPRRRRQRRRQSRSQRSRQQTRQQSRSPQSQQPRRSQQQSRSRTRRSQRQQTRSQQRRSPQQRPQSQRPQSSRQQQSQSSSRRRSRQSRRRSRRRSRRSRPQQQSRRQRSRRTRRQPQSRQRQSRTTTRPRQPQQTRSRRRRS

SRSQQTQSRSRRRPRQQRRQQRQQQQRQPQQSQRUQQSQRRSQRSSSQQQQSQQQQSQSSSQQPPRSPPRQSQSSSSPRRQRQPQQQRSQRQRQQQQQRPRRRRPQRQQQSQQQRPRRRSSRRSRTSRRQRQQPQQRQQQRRSRQQSRSQQQRQRQQQQQQQQQQTSRRRSRRRRQQQQTQPTPQQSRR

RQQRQQSQQQQPQQQRRRQQQQQRSRPQRPRSQQQRRRSRRQSPSQRSSQSRRRQQSRRSQPRRQRQSRRQRRQSRQQUQTQPRQPQRQQQRRQSQQQRQTTPQRTRRPPQRRRQQSQRRRRQRRQRRRRSQPPRRRQUQSQRRRVVSRQSTRRRTRRQPRQQQRRVSRSRRSQRSQQRQSQRRTSQQS

TPSRSSRRTRRRRPRRQQQQRTPQQQPQQSQQQRRRRQTRRRQRRTQUSSRQRSSSRSQRQRRRRRRRTQQQTRTVQQQQQQRTPQQQPQQSRQQSPRRQRRPSRQQRQQTTRTQQPQQQTQQRQRPPQQRQRQRSQQRRQRQRRTTQRTSQQRSRRSRRTURRRQQSRRRQQQRRSRRSRSRRRQQQQ

QRRSQQPSQRRQRRQPQQQRRSRRQPQSSSRTQQQTQPRQQQPSSRQQRQRQPQRRSSRRQPQSRQSRQQRRRRRRQRRRRRQQQRQRSQRQQQRRRRQRRQRQQPSQQVRQRRQQSRPSSRSRRRRQTSSQQRRQRRRPQRQRRQTRRSRRRRQQQQRQTRSQRRRQPQQQQQRSRRRQQRQQQQQPQ

QRRTRPRQSRRRQRQPQRQQQSSRSTQRQRRPRQPQQSQRSRRRSRSSRSRTQQQQRQRQQQQQSSQRRQRVRRQRRTSRQRRQQPQPQRSQRSTRQRRQPQQSUQQTQRQRRRPQSQQQRRQQQQRTQRPRQQQRRQQTSPSQQRQRRQRRRURRRRQRPRQQRQRQPQQQQRSQQQQQRQTTQQQQQ

QQQTSRSRQPRRSQQRQQTQQSRQQRQRSSSQQRRRQQRTRSQRSSQQPSQQQRPQRPTRQQRUQTSSQRRSRTRRQRRRRSRQRRPPSRSRRQRTRRRRQRRQQRRRRQRSRQSQQTQRQQRQRQRSQSRQRQQSRRRRQQQSQRRRSQRSRQRTURQQSRQQQRQRRQRQQQSRRQRRPQRSRRRRQ

SQTQSSRQQUSQRPQSRPQRRRRQRRSTRSQQPQTQSQSQQPQRRQRPRURRSRQSQQTRRSSSRSSRRRQRQQRRRRSSRRSSQTRRRQSSRQQSSSQRRRRRRRRQQSQRSRRSQRQSQQQRRSQQRRTRRRQQRQPRRQSSRRQQTRVPRUQQPRSQPRRSSQQRQQSRQSRTSRRQRQRRSRRQR

RSRRRQSRSRRRRSTRQRSRRRRSQQQVRRRQRQQRURSQRRRQRQRRQQSRPRRRQPSQSRRPSSQRRQQSQRSRQSQRSRSQRQQPRSSQSPRRQTQQQRPQQRTSRQQQRQRRRRSQSSQSRQQURURRPSRRQPRRQQRRRRQRPQQSQRRQRRSRSSRQRSRRRSRRSQQURRSTQQQTSQQRR

QTRURQSRQRQSQQQQSSSRQQRPTRQQRQSQQQTQRQRSRQRPPSRSSQQQRSRRRSQRRSQRRSQQRQTSQQRTPRSQQSSQQRQRPRSRRURSQSSRQSPQRRQSRQSPRTTQTSSRTSRRRTSRRQQQQRQQQPPRQRQSRRQQQTQTRRPRPRQQQQRRQQRQRRQQSQRQSRQRSQQRTUQQS

QRRRTRQSURQQQQRSQSTRRQQQTRRTQRSPSPRRQRSRQQQRRRQQQVUTSQRSPRRRRQRRSQRQQQRPQQQRRQSTRRQSRQRRSQSPRQRSQPQQQQQRSRRSSRPQRTSSRRSPQQSRRTRPRRQTRQQQQSRVRPRSSQQRQRRQRRRQQRQRQQRPRPRRPPRQQQRSSQRQQSSTRQQPQ

QRRRQQPRTTRRPRRQRRURTQQQQRQQQPPPRSRQSRQRSQQRQTPTQQRQQSRUTRRSQQQQPSRPPPRQQQRTPRQQQSRRRQRQTQRRQSSSTSRRSRRQSPQQQQRQQQQTSQTRPSRRRSTRTPQRSRQSRUQUS9QQRQPQQ)3!!!!!!!!
MD:Z:22^A2757^A25T327^C10 NH:i:1 HI:i:1 NM:i:5 SM:i:40 XQ:i:40 X2:i:0
XO:Z:UU PG:Z:M


----------------------


Then I ran the sort command:

sort -k 3,3 -k 4,4n all_quivered_hq.100_30_0.99.sam >
all_quivered_hq.100_30_0.99.sorted2.sam


Then the output file all_quivered_hq.100_30_0.99.sorted2.sam:


The header is missing in the header. Do you know why?


@HD VN:1.0 SO:unsorted

@PG ID:GMAP PN:gmap VN:2014-06-10 CL:gmap -D
/global/homes/j/jianpeng/Reference_genome/Athaliana/referenceGenome -d
arabidopsis -f samse -n 0 -t 24 --min-trimmed-coverage 0.99 --min-i

dentity 0.85 all_quivered_hq.100_30_0.99.fastq

@SQ SN:Chr1 LN:30427671

@SQ SN:Chr2 LN:19698289

@SQ SN:Chr3 LN:23459830

@SQ SN:Chr4 LN:18585056

@SQ SN:Chr5 LN:26975502

@SQ SN:ChrC LN:154478

@SQ SN:ChrM LN:366924

------------------------

**Comment by Magdoll** [comment]
Hi,

I believe this is the same error as [this bug](https://github.com/PacificBiosciences/cDNA_primer/issues/2). The fix appears to be getting pbcore installed to 0.9.1 version. PLease try it and let me know if it works.

Thanks,
--Liz

**Comment by jxu006** [comment]
Hi Liz,

It seems that my pbcore is 0.9.0.

I downloaded the wget --trust-server-names
https://github.com/PacificBiosciences/pbcore/archive/0.9.1.tar.gz

I used the command according the README:

jixu@gpint100:~/pbcore-0.9.1$ pip install nump

Downloading/unpacking nump

  Could not find any downloads that satisfy the requirement nump

Cleaning up...

No distributions at all found for nump

Storing debug log for failure in /global/homes/j/jianpeng/.pip/pip.log


Do you know what is going on? How to fix this?


Thanks,


Jianpeng

**Comment by adeslatt** [comment]
Hi Jack,
You can get pbcore 0.9.1 here under the pbcore, look under releases.
https://github.com/PacificBiosciences/pbcore/archive/0.9.1.tar.gz
Anne

**Comment by jxu006** [comment]
Hi Liz,

I have a question:
Which file can I find the number of full pass information? I want to find
the number of full pass information, can you give me a little instruction
on this?

Thanks a lot,

Jack

**Comment by Magdoll** [comment]
Hi Jack,

Do you mean:

(1) For each ReadsOfInsert read, the number of passes in that read? 

or

(2) For each Iso-Seq "cluster" output (Quiver HQ or Quiver LQ), the number of reads belonging to that cluster?

**Comment by jxu006** [comment]
I mean the first one:

(1) For each ReadsOfInsert read, the number of passes in that read? I have
the reads_of_insert.fasta file, I do not see the numbers of pass in this
file. Any other file should I go ?

Thanks,

Jack

**Comment by Magdoll** [comment]
Hi Jack,

That information is kept in the .ccs.h5 files. You would have to get the <movie_name>.ccs.h5 files and have pbcore installed.

Here's some example code:
```python

from pbcore.io import BasH5Reader
bas = BasH5Reader('m150125_010339_42161_c100730732540000001823150705141525_s1_p0.3.ccs.h5')

holes = bas.sequencingZmws

x = bas[108988]
x.hqRegion, x.productivity, x.numPasses,

**Comment by jxu006** [comment]
Thanks. I tried to run tofu_wrap.py, but it showed:


> tofu_wrap.py -h


Traceback (most recent call last):

  File "/global/homes/j/tofu/cDNA_primer/bin/tofu_wrap.py", line 5, in
<module>

    pkg_resources.run_script('pbtools.pbtranscript==0.4.1', 'tofu_wrap.py')

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pkg_resources.py",
line 534, in run_script

    self.require(requires)[0].run_script(script_name, ns)

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pkg_resources.py",
line 1434, in run_script

    execfile(script_filename, namespace, namespace)

  File
"/global/u2/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/EGG-INFO/scripts/tofu_wrap.py",
line 6, in <module>

    from pbtools.pbtranscript.Cluster import Cluster

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/Cluster.py",
line 11, in <module>

    from pbtools.pbtranscript.Polish import Polish

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/Polish.py",
line 13, in <module>

    from pbtools.pbtranscript.ice.IceUtils import build_sa, \

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/ice/IceUtils.py",
line 19, in <module>

    from pbtools.pbtranscript.io.BasQV import basQVcacher

  File
"/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/io/BasQV.py",
line 45, in <module>

    import pbtools.pbtranscript.io.c_basQV as c_basQV

ImportError: /lib/libc.so.6: version `GLIBC_2.14' not found (required by
/global/homes/j/tofu/cDNA_primer/lib/python2.7/site-packages/pbtools.pbtranscript-0.4.1-py2.7-linux-x86_64.egg/pbtools/pbtranscript/io/c_basQV.so)


Do you know what is going on ? How to fix this?


Thanks,

**Comment by Magdoll** [comment]
This looks like it might be coming from different environments. Did you use the SMRTAnalysis 2.3 as the basis for the TOFU install? 

You may also want to look into actually getting glibc installed if it's missing. See this thread:
http://askubuntu.com/questions/421642/libc-so-6-version-glibc-2-14-not-found

**Comment by jxu006** [comment]
The tofu was installed under SMRTAnalysis 2.3.

**Comment by Magdoll** [comment]
Then you may want to see if you can fix the lack of libc.so.6 using the link above.

**Comment by jxu006** [comment]
​
 all_quivered_hq.100_30_0.99.sam
<https://docs.google.com/file/d/0B8aYrEKcOs6eQm1iUUlzZlpmYkk/edit?usp=drive_web>
​​
 combined.all_quivered_hq.100_30_0.99.fasta
<https://docs.google.com/file/d/0B8aYrEKcOs6eTG50ZG9nV0NWNTg/edit?usp=drive_web>
​Hi Liz,

I do not have the administer access to my smrtanalysis 2.3.0 in my cluster.
I have informed the administer to fix this error about the pbcore 0.9.1.
But it need some time to fix it.

I was wondering whether I can send my fasta and SAM file to you and you can
help run the collapse redundancy transcripts and send the results back to
me.

Thanks a lot,

Jack

**Comment by jxu006** [comment]
Hi Liz,


I ran the ConsensusTools.sh using two different commands:


1. ConsensusTools.sh CircularConsensus  --minFullPasses 0
--minPredictedAccuracy 75 --parameters
/global/projectb/shared/software/smrtanalysis/2.3.0/current/analysis/etc/algor

ithm_parameters/2014-09 --numThreads 24 --fofn input.fofn -o output


2. ConsensusTools.sh CircularConsensus  --minFullPasses 1
--minPredictedAccuracy
75 --parameters
/global/projectb/shared/software/smrtanalysis/2.3.0/current/analysis/etc/algor

ithm_parameters/2014-09 --numThreads 24 --fofn input.fofn -o output


The only difference for these two runs is the parameter --minFullPasses. One
is 0 and the other run is 1. Then I generate the *.ccd.h5 file.


Then I downloaded the ccs_passes.py according to the link:


https://github.com/PacificBiosciences/Bioinformatics-Training/wiki/Extracting-Reads-of-Insert-(CCS)-number-of-full-passes


Then I calculated the number of passes using ccs_passes.py.


The results showed that the minimum number of full pass for the *.ccs.h5
generated from --minFullPasses 0 is 1. The minimum number of full pass for the
*.ccs.h5 generated from --minFullPasses 1 is 2.


If I specify --minFullPasses 0, why the minimum number of full pass for the
*.ccs.h5 is 1? Should it be 0?

If I specify --minFullPasses 1, why the minimum number of full pass for the
*.ccs.h5 is 2? Should it be 1?


Can you give me a little explanation?


Thanks a lot,


Jack
